apps/okreads-e2e/src/specs/search-books.spec.ts

```diff
@@ -16,12 +16,58 @@ describe('When: Use the search feature', () => {
     expect(items.length).toBeGreaterThan(1);
   });
 
-  xit('Then: I should see search results as I am typing', async () => {
+  it('Then: I should see search results as I am typing', async () => {
     await browser.get('/');
     await browser.wait(
       ExpectedConditions.textToBePresentInElement($('tmo-root'), 'okreads')
     );
 
-    // TODO: Implement this test!
+    const input = await $('input[type="search"]');
+
+    await input.sendKeys('ja');
+    // search term is 'ja'
+
+    const jaItems = await $$('[data-testing="book-item"]');
+    expect(jaItems.length).toBeGreaterThan(1);
+
+    await input.sendKeys('v');
+    // search term is now 'jav'
+    const javItems = await $$('[data-testing="book-item"]');
+    expect(javItems.length).toBeGreaterThan(1);
+
+    // results have been updated accordingly
+    expect(jaItems === javItems).toEqual(false);
+
+    await input.sendKeys('as');
+    // search term is now 'javas'
+    const javasItems = await $$('[data-testing="book-item"]');
+    expect(javasItems.length).toBeGreaterThan(1);
+
+    await input.sendKeys('cript');
+    // search term is now 'javascript'
+    const javascriptItems = await $$('[data-testing="book-item"]');
+    expect(javascriptItems.length).toBeGreaterThan(1);
+
+    await input.sendKeys(Key.BACK_SPACE);
+    await input.sendKeys(Key.BACK_SPACE);
+    // search term is now 'javascri'
+    const javascriItems = await $$('[data-testing="book-item"]');
+    expect(javascriItems.length).toBeGreaterThan(1);
+
+    await input.sendKeys(Key.BACK_SPACE);
+    await input.sendKeys(Key.BACK_SPACE);
+    // search term is now 'javasc'
+    await input.sendKeys(Key.BACK_SPACE);
+    const javascItems = await $$('[data-testing="book-item"]');
+    expect(javascItems.length).toBeGreaterThan(1);
+    // search term is now 'javas'
+    const javasItems2 = await $$('[data-testing="book-item"]');
+    expect(javasItems2.length).toBeGreaterThan(1);
+
+    await input.sendKeys(Key.BACK_SPACE);
+    // search term is now 'java'
+    const javaItems = await $$('[data-testing="book-item"]');
+    expect(javaItems.length).toBeGreaterThan(1);
+    expect(javaItems === javascriItems).toEqual(false);
```

Comment:
This could all be simplified greatly. What you need to check is that the books area loaded without the user having to press enter or click a button. You can achieve this by sending keys to the input control and then awaiting for `$$('[data-testing="book-item"]')` to be populated.